internal/evmreader/evmreader.go

```diff
@@ -39,6 +40,16 @@ type EvmReaderRepository interface {
 	GetAllRunningApplications(ctx context.Context) ([]Application, error)
 	GetNodeConfig(ctx context.Context) (*NodePersistentConfig, error)
 	GetEpoch(ctx context.Context, indexKey uint64, appAddressKey Address) (*Epoch, error)
+	GetPreviousSubmittedClaimsEpochs(
```

Comment:
The current name is quite odd. I would rather choose something like `GetEpochsWithOpenClaims` or `GetEpochsWithUnfinishedClaims`, as "previously submitted claims" is an ambiguous term that includes approved or rejected claims.